qpid-jms-client/src/main/java/org/apache/qpid/jms/provider/amqp/AmqpSaslAuthenticator.java

```diff
@@ -137,7 +142,9 @@ private void handleSaslStep() throws JMSSecurityException {
                 byte[] challenge = new byte[sasl.pending()];
                 sasl.recv(challenge, 0, challenge.length);
                 byte[] response = mechanism.getChallengeResponse(challenge);
-                sasl.send(response, 0, response.length);
+                if (response != null) {
```

Comment:
Good point. Interesting, I guess it is allowing for the 'additional-data' style information at the end of the exchange where there isn't a response.